### PR TITLE
#6230 add missing update_seq option to allDocs() and query()

### LINKED ISSUE #6230: Missing update_seq option
CouchDB has support for the `update_seq` option in e.g. [all_docs](http://docs.couchdb.org/en/2.0.0/api/database/bulk-api.html).

This seems to be missing in PouchDB?

### PR DESCRIPTION
**Pull request by dharders**
Closes #6230 

Adds the missing `update_seq` option to the `allDocs()` and `query()` api's.

i.e. You can now:

```javascript
db.allDocs({update_seq: true}).then(result => { 
   // result.update_seq is set correctly
}

db.query({update_seq: true}).then(result => { 
   // result.update_seq is set correctly
}
```

The `update_seq` value respects the adapter format. i.e. integer or string (eg '4-XXXXXXXXXX')

**Adapters updated:**

- http
- indexeddb
- idb
- leveldb
- websql

API documentation updated
Tests included

**References**
[http://docs.couchdb.org/en/2.1.1/api/database/bulk-api.html#get--db-_all_docs](http://docs.couchdb.org/en/2.1.1/api/database/bulk-api.html#get--db-_all_docs)
[http://docs.couchdb.org/en/2.1.1/api/ddoc/views.html#db-design-design-doc-view-view-name](http://docs.couchdb.org/en/2.1.1/api/ddoc/views.html#db-design-design-doc-view-view-name)

Note: I haven't included any 'istanbul ignores' yet so I expect coverage to drop. But just wanted to kick off the other tests.

### CODE REVIEW

**Comment by daleharvey** [comment]
Hey @dharders so the conflict is your other PR that landed disabling the test, I went through this PR and everything looks great, I think this will be good to land once rebased and green :)

**Comment by dharders** [comment]
Hey @daleharvey everything is now green except coverage.

Any hints to fix this? I.e. Is this just Istanbul ignores need to be added or am I actually missing some tests?

Coverage seems to be 9 "lines" short. How do I find out which lines from the log?

Ps how do I squash all the duplicate commits from the rebase? (I'm not super familiar with git)

**Comment by ptitjes** [comment]
@dharders After making sure you master branch is up-to-date, you have to do
```
git checkout 6230-add-alldocs-update-seq-option
git rebase master
# Resolve conflicts now, if any
# And then
git push -f origin 6230-add-alldocs-update-seq-option
```

Note the `-f` on the push. It force your remote `6230-add-alldocs-update-seq-option` branch to be the one you have locally. Currently, you must be using merge, or something, but the result is that you still carry all the commits made by others since you started your PR. (and don't worry GitHub handles the force push very well.)

**Comment by dharders** [comment]
Thanks for the info @ptitjes ;)

Any ideas on fixing the coverage?

**Comment by ptitjes** [comment]
@dharders Nope, no idea... Sorry.

**Comment by daleharvey** [comment]
So if you run the test coverage locally then it will give you a html output that will point to the exact lines that are missed, If I get some time I will try to do it tomorrow

**Comment by dharders** [comment]
Ahh cool. I think I skipped this step as I was developing on win10 (Linux subsystem) and had an issue with a dependency. I might fire up a vm and check it out. Thanks for your patience and guidance!

**Comment by dharders** [comment]
All green @daleharvey ;)

**Comment by daleharvey** [comment]
So in theory we should be testing against a pouchdb-server that includes the up to date pouchdb, which should support update_seq so the test skips shouldnt be needed, but its possible we change that and I missed it, so happy to merge this but will file a follow up that we get this supported in pouchdb-server and can delete the isPouchServer and the istanbul if's

**Comment by dharders** [comment]
Yep, agree about deleting isPouchServer() and Istanbul if’s. They are only required until pouchdb-server is updated with pouchdb master (or next release version). It was a catch 22 as this PR added the update_seq functionality that we were testing pouchdb-server for, which was yet to be merged as it was 1 version behind this (yet to be released) master version. If that makes sense...?

Thanks again for walking me through this!

**Comment by daleharvey** [comment]
So it was actually a bug that was found in https://github.com/pouchdb/pouchdb/pull/6858, for this exact reason we link the newest version of pouchdb to pouchdb server when we run the tests however that linking got broken at some point

**Comment by dharders** [comment]
Ahhh, good catch. Makes sense now. Once the `bin/run-tests.sh` script is updated (and npm link to PR derived pouchdb-server) the workarounds should be able to be deleted safely and hopefully still pass and have full coverage. 🤞

File: packages/node_modules/pouchdb-adapter-indexeddb/src/allDocs.js [hunk 1]
```diff
@@ -37,11 +37,16 @@ function createKeyRange(start, end, inclusiveEnd, key, descending) {
...
       total_rows: metadata.doc_count,
       offset: opts.skip,
       rows: []
-    });
+    };
+    /* istanbul ignore if */
+    if (opts.update_seq) {
+      returnVal.update_seq = metadata.seq;
+    }
+    callback(null, returnVal);
```
Comment (nename0) [inline packages/node_modules/pouchdb-adapter-indexeddb/src/allDocs.js]: Hey @dharders you missed a `return` there. When running the integration test I noticed this. (Sadly it's not so straightforward to do so with the `indexeddb `adapter as there is no query parameter. You need to manualy change some stuff)

Comment (dharders) [reply packages/node_modules/pouchdb-adapter-indexeddb/src/allDocs.js]: Yep, you're right. Fixed in #6864 (I must have missed this after rebase manual changes I made)

**Pull request merged**